commands/commands.go

```diff
@@ -141,15 +132,49 @@ func RunWithTimeout(c *Command, fields log.Fields) error {
 	log.Debugf("%s.RunWithTimeout start", c.Name)
 	c.setUpCmd(fields)
 	defer c.closeLogs()
-	log.Debugf("%s.Cmd.Start", c.Name)
+	log.Debugf("%s.Cmd start", c.Name)
+	ctx, cancel := context.WithTimeout(context.Background(), c.TimeoutDuration)
+	defer cancel()
 	if err := c.Cmd.Start(); err != nil {
-		log.Errorf("Unable to start %s: %v", c.Name, err)
+		log.Errorf("unable to start %s: %v", c.Name, err)
+		return err
+	}
+	go func() {
+		select {
+		case <-ctx.Done():
+			// if the context was canceled we don't want to kill the
+			// process because it's already gone
+			if ctx.Err().Error() == "context deadline exceeded" {
```

Comment:
Yeah this is a very unfortunate design choice in the golang stdlib. There are very few cases where there's an error type wrapping `error` that you can assert types on, as they think this is an anti-pattern (wtf?). Fortunately we have several unit tests that will break if they change the text, so I think we're ok here.